src/redux/movies.ts

```diff
@@ -0,0 +1,121 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import type { PayloadAction } from '@reduxjs/toolkit';
+import { Movie } from 'types/movie';
+import { RootState } from './store';
+
+interface MoviesState {
+  movies: Movie[];
+  selectedMovie?: Movie;
+  moviesCount: number;
+}
+interface fetchProps {
+  searchText?: string;
+  searchBy?: string;
+  sortBy?: string;
+  sortOrder?: string;
+  filter?: string[];
+}
+const initialState: MoviesState = {
+  movies: [],
+  selectedMovie: undefined,
+  moviesCount: 0,
+};
+export const fetchMoviesAction = createAsyncThunk(
+  'moviesFetch',
+  (params: fetchProps) => {
+    return fetchMovies(params);
+  },
+);
+
+const fetchMovies = async ({
+  searchText,
+  searchBy,
+  sortBy,
+  sortOrder = 'asc',
+  filter,
+}: fetchProps) => {
+  const queryParams = [
+    searchText ? `search=${searchText}` : '',
+    searchBy ? `&searchBy=${searchBy}` : '',
+    sortBy ? `&sortBy=${sortBy}` : '',
+    filter?.length ? `&filter=${filter.join()}` : '',
+    `sortOrder=${sortOrder}`,
+  ]
+    .filter(Boolean)
+    .join('&');
+  const response = await fetch(`http://localhost:4000/movies?${queryParams}`);
```

Comment:
Never hardcode any base URL's domains in your code. Usually apps are migrating from one domain to another, to make it easier to support, use `.env` file to storing all app env variables (URLs, creds and etc). Take a look at my example [here](https://github.com/dotnet-fizzyy/Diploma/blob/V2/web/.env) how to store